### PR TITLE
Functional tests - Product settings display remaining quantity in product page

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Test display remaining quantity in product page
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | TEST_PATH="functional/BO/13_shopParameters/03_productSettings/02_productPage/02_displayRemainingQuantities.js" URL_FO=yourShopURL npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17754)

### CODE REVIEW

File: tests/puppeteer/pages/FO/product.js [hunk 1]
```diff
@@ -45,4 +46,12 @@ module.exports = class Product extends FOBasePage {
     await this.waitForSelectorAndClick(this.productDetail);
     return this.elementVisible(this.productQuantitySpan, 1000);
   }
+
+  /**
+   * Is availability product displayed
+   * @returns {boolean|true}
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/FO/product.js]:
```suggestion
   * @returns {boolean}
```

File: tests/puppeteer/pages/FO/searchResults.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+require('module-alias/register');
+const FOBasePage = require('@pages/FO/FObasePage');
+
+module.exports = class SearchResults extends FOBasePage {
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/FO/searchResults.js:4]:
```suggestion
module.exports = class SearchResultsPage extends FOBasePage {
```

File: tests/puppeteer/pages/FO/searchResults.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  /**
+   * Go to the product page
+   * @param id, product id
+   */
+  async goToProductPage(id) {
+    await Promise.all([
+      this.page.waitForNavigation({waitUntil: 'networkidle0'}),
+      this.page.click(this.productImg.replace('%NUMBER', id)),
+    ]);
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/FO/searchResults.js]:
```suggestion
    await this.clickAndWaitForNavigation(this.productImg.replace('%NUMBER', id));
```

File: tests/puppeteer/campaigns/data/faker/product.js [hunk 4]
```diff
@@ -6,7 +6,7 @@ module.exports = class Product {
     this.summary = productToCreate.summary === undefined ? faker.lorem.sentence() : productToCreate.summary;
     this.description = productToCreate.description === undefined ? faker.lorem.sentence() : productToCreate.description;
     this.reference = faker.random.alphaNumeric(7);
-    this.quantity = productToCreate.quantity || faker.random.number({min: 1, max: 9}).toString();
+    this.quantity = productToCreate.quantity.toString() || faker.random.number({min: 1, max: 9}).toString();
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/data/faker/product.js]:
```suggestion
    this.quantity = (productToCreate.quantity || faker.random.number({min: 1, max: 9})).toString();
```

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/02_displayRemainingQuantities.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
+require('module-alias/register');
+const testContext = require('@utils/testContext');
+
+const baseContext = 'functional_BO_productSettings_displayAvailableQuantitiesOnProductPage';
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/02_displayRemainingQuantities.js]:
```suggestion
const baseContext = 'functional_BO_productSettings_displayRemainingQuantities';
```

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/02_displayRemainingQuantities.js [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    await expect(validationMessage).to.equal(this.pageObjects.addProductPage.settingUpdatedMessage);
+  });
+
+  it('should go to \'Shop parameters > Product Settings\' page', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'goToProductSettingsPage', baseContext);
+    await this.pageObjects.boBasePage.goToSubMenu(
+      this.pageObjects.boBasePage.shopParametersParentLink,
+      this.pageObjects.boBasePage.productSettingsLink,
+    );
+    await this.pageObjects.boBasePage.closeSfToolBar();
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/02_displayRemainingQuantities.js]:
```suggestion
```

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/02_displayRemainingQuantities.js [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+
+  // Login into BO and go to product settings page
+  loginCommon.loginBO();
+
+  it('should go to \'Catalog > Products\' page', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'goToProductsPage', baseContext);
+    await this.pageObjects.boBasePage.goToSubMenu(
+      this.pageObjects.boBasePage.catalogParentLink,
+      this.pageObjects.boBasePage.productsLink,
+    );
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/02_displayRemainingQuantities.js:69]:
```suggestion
    );
    await this.pageObjects.boBasePage.closeSfToolBar();
```

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/02_displayRemainingQuantities.js [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+    await this.pageObjects.homePage.searchProduct(productData.name);
+    await this.pageObjects.searchResultsPage.goToProductPage(1);
+    const AvailableQuantityIsVisible = await this.pageObjects.productPage.isAvailabilityQuantityDisplayed();
+    await expect(AvailableQuantityIsVisible).to.be.false;
+    page = await this.pageObjects.productPage.closePage(browser, 1);
+    this.pageObjects = await init();
+  });
+
+  it('should go to \'Shop parameters > Product Settings\' page', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'goToProductSettingsPage', baseContext);
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/02_displayRemainingQuantities.js]: please modify the test Identifier, it's a duplicate from line 82

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/02_displayRemainingQuantities.js [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+  });
+
+  it('should update Display remaining quantities to the default value', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'setDisplayRemainingQuantityDefaultValue', baseContext);
+    const result = await this.pageObjects.productSettingsPage.setDisplayRemainingQuantities(defaultRemainingQuantity);
+    await expect(result).to.contains(this.pageObjects.productSettingsPage.successfulUpdateMessage);
+  });
+
+  it('should check that the product availability is displayed in FO product page', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'checkRemainingQuantityAlert', baseContext);
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/02_displayRemainingQuantities.js]: Same here, il's a duplicate from line 99

File: tests/puppeteer/pages/FO/searchResults.js (hunk 2 above)
Comment (nesrineabdmouleh) [reply tests/puppeteer/pages/FO/searchResults.js:4]: I think that in all the pages, the class name does not contain Page.

Comment (boubkerbribri) [reply tests/puppeteer/pages/FO/searchResults.js:4]: Ok :+1:

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/02_displayRemainingQuantities.js [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+    await expect(result).to.contains(this.pageObjects.productSettingsPage.successfulUpdateMessage);
+  });
+
+  it('should check that the product availability is not displayed in FO product page', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'checkThatRemainingQuantityAlertNotVisible', baseContext);
+    page = await this.pageObjects.productSettingsPage.viewMyShop();
+    this.pageObjects = await init();
+    await this.pageObjects.homePage.searchProduct(productData.name);
+    await this.pageObjects.searchResultsPage.goToProductPage(1);
+    const AvailableQuantityIsVisible = await this.pageObjects.productPage.isAvailabilityQuantityDisplayed();
```
Comment (Progi1984) [inline tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/02_displayRemainingQuantities.js]:
```suggestion
    const availableQuantityIsVisible = await this.pageObjects.productPage.isAvailabilityQuantityDisplayed();
```

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/02_displayRemainingQuantities.js [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+  });
+
+  it('should check that the product availability is displayed in FO product page', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'checkThatRemainingQuantityAlertIsVisible', baseContext);
+    page = await this.pageObjects.productSettingsPage.viewMyShop();
+    this.pageObjects = await init();
+    await this.pageObjects.homePage.searchProduct(productData.name);
+    await this.pageObjects.searchResultsPage.goToProductPage(1);
+    this.pageObjects = await init();
+    const AvailableQuantityIsVisible = await this.pageObjects.productPage.isAvailabilityQuantityDisplayed();
```
Comment (Progi1984) [inline tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/02_displayRemainingQuantities.js]:
```suggestion
    const availableQuantityIsVisible = await this.pageObjects.productPage.isAvailabilityQuantityDisplayed();
```

**Review by boubkerbribri — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Pull request merged**